client/client-multi/client-cpp/CMakeLists.txt

```diff
@@ -393,6 +393,7 @@ endif()
 find_package(Boost 1.54 REQUIRED COMPONENTS log system thread)
 find_package(Avro REQUIRED)
 find_package(Botan REQUIRED)
+find_package(Threads REQUIRED)
```

Comment:
Indeed, the `find_package(Boost 1.54 REQUIRED COMPONENTS log system thread)` successfully finds the **boost.threads** library, but don't confuse it with **Threads** package which provides platform-specific threading implementation. Initially it was expected that the above command will take care of finding the Threads package and initialize the `CMAKE_THREAD_LIBS_INIT` CMake variable, however this does not seem to be true for all platforms.